mhs/common/mhs_common/workflow/asynchronous_express.py

```diff
@@ -87,5 +90,18 @@ def _record_outbound_audit_log(self, end_time, start_time, acknowledgment):
         await wdo.set_status(wd.MessageStatus.OUTBOUND_MESSAGE_PREPARED)
         return None, message
 
-    async def handle_inbound_message(self, work_description: wd.WorkDescription, payload: str):
-        pass
+    @timing.time_function
+    async def handle_inbound_message(self, message_id: str, correlation_id: str, work_description: wd.WorkDescription,
+                                     payload: str):
+        logger.info('0009', 'Entered async express workflow to handle inbound message')
+        await work_description.set_status(wd.MessageStatus.INBOUND_RESPONSE_RECEIVED)
+        try:
+            await self.queue_adaptor.send_async(payload, properties={'message-id': message_id,
+                                                                     'correlation-id': correlation_id})
```

Comment:
Here I'm passing the message id & correlation id as [additional parameters](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-application-properties). The AMQP spec does have message & correlation ids as part of the [standard properties](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-properties), but it says in the spec that the message id should be unique, otherwise messages may get discarded. And I'm not sure we can say that with certainty. Then I didn't want to put the correlation id as a standard parameter and have the message id as an additional property, so I just made them both additional properties.

@gareth-allan @dshipp-bjss any thoughts on this?

Reply:
If we're using the message ID (rather than the conversation ID), my understanding is that it's intended to identify an individual message, so should be unique. They're also UUIDs, so uniqueness shouldn't be an issue from that point of view. Is the question whether a retried messages will contain a new message ID?

Reply:
Between @briandiggle, me and @gareth-allan, we've discussed this a bit. The conclusion was to keep this as-is, it's simple enough as it is, possibly we should rename `message-id` to something else, but I'd want this to be discussed/thought through further, outside of this PR, before making any changes to this.